osu.Game/Screens/Select/Details/AdvancedStats.cs

```diff
@@ -65,12 +60,60 @@ public AdvancedStats()
                     starDifficulty = new StatisticRow(10, true) { Title = "Star Difficulty" },
                 },
             };
+            
         }
 
         [BackgroundDependencyLoader]
-        private void load(OsuColour colours)
+        private void load()
         {
             starDifficulty.AccentColour = colours.Yellow;
+            mods.ValueChanged += _ => updateStatistics();
+        }
+
+        private void updateStatistics()
+        {
+            BeatmapInfo processed = Beatmap?.Clone();
+
+            if (processed != null && mods.Value.Any(m => m is IApplicableToDifficulty))
+            {
+                processed.BaseDifficulty = processed.BaseDifficulty.Clone();
```

Comment:
is it necessary to clone a second time?

Reply:
For some reason we need to get a shallow clone of `BeatmapDifficulty`. Certain issues such as the bars not getting colored and the difficulty not showing correct values when re-entering song select will show up.

It is also done in `WorkingBeatmap`:https://github.com/ppy/osu/blob/0627c814d66c34925057fa38a5dc50bd33144d5e/osu.Game/Beatmaps/WorkingBeatmap.cs#L124-L125